openapi/templates/index.js

```diff
@@ -474,3 +476,21 @@ function returnsPolicyRule(operations) {
     }).length > 0
   );
 }
+
+const URN_GRANT_TYPES = ['device_code', 'saml2_bearer', 'token_exchange'];
+const URN_PREFIX = 'urn:ietf:params:oauth:grant-type:';
+const URN_SUFFIXES = {
+  'device_code': 'device_code',
+  'saml2_bearer': 'saml2-bearer',
+  'token_exchange': 'token-exchange'
+};
+
+// Determines if OAuthGrantType needs prefix
+function isUrnGrantType(grantType) {
+  return URN_GRANT_TYPES.includes(grantType);
+}
+
+// Prefixes openAPI spec enum with OAuthGrantType urn prefix
+function prefixUrnGrantType(grantType) {
+  return URN_PREFIX + URN_SUFFIXES[grantType];
+}
```

Comment:
Helper functions used to determine if a grant type needs the urn prefix (`urn:ietf:params:oauth:grant-type:`) and if so, to prefix it onto the actual enum value